telegram/_bot.py

```diff
@@ -2037,16 +2049,52 @@ async def send_media_group(
                 :attr:`~telegram.request.BaseRequest.DEFAULT_NONE`.
             api_kwargs (:obj:`dict`, optional): Arbitrary keyword arguments to be passed to the
                 Telegram API.
+            caption (:obj:`str`, optional): Caption that will be added to the
+                first element of :paramref:`media`, so that it will be used as caption for the
+                whole media group.
+                Defaults to :obj:`None`.
+            parse_mode (:obj:`str` | :obj:`None`, optional):
+                Parse mode for :paramref:`caption`.
+                See the constants in :class:`telegram.constants.ParseMode` for the
+                available modes.
+                Defaults to :obj:`None`.
+            caption_entities (List[:class:`telegram.MessageEntity`], optional):
+                List of special entities for :paramref:`caption`,
+                which can be specified instead of :paramref:`parse_mode`.
+                Defaults to :obj:`None`.
 
         Returns:
             List[:class:`telegram.Message`]: An array of the sent Messages.
 
         Raises:
             :class:`telegram.error.TelegramError`
         """
+        if caption and any(
+            [
+                any(item.caption for item in media),
+                any(item.caption_entities for item in media),
+                any(item.parse_mode is DEFAULT_NONE for item in media),
```

Comment:
```suggestion
                any(item.parse_mode is not DEFAULT_NONE for item in media),
```

`DEFAULT_NONE` is the value that we expect, as it indicates that the `parse_mode` was not manually specified :)

Reply:
Well, you're absolutely right that I forgot `not` there, I was just lucky `ValueError` was thrown because of preceding statements.

But I can't put `is not DEFAULT_NONE` either because then the test will fail, and in the test the test media group has media items with `parse_mode` explicitly set to None.

So I guess the line should say, just like in preceding lines:
```py
any(item.parse_mode for item in media)
```
Because who knows, not only in test can we explicitly pass `None` to `parse_mode`, maybe some user will do that too for some reason.

Reply:
Then the unit test needs to be adapted ;)

We want to check if the user explicitly set the `parse_mode` for the `InputMedia*` object. That's the case, if `item.parse_mode is not DEFAULT_NONE`. If it's `None`, then the user explicitly did `InputMediaPhoto(…, parse_mode=None)` and that's a case where we don't want to allow passing `send_media_group(…, caption='stuff')`.

Reply:
I see. Let me go a bit back then and ask: how is `parse_mode` different from `caption` and `caption_entities`? Why, say, in the definition of `send_audio()` does `parse_mode` have `= DEFAULT_NONE`, while `caption` and `caption_entities` have `= None`? I think I'm especially curious as to how `parse_mode` is different from `caption_entities`

Reply:
`parse_mode` differs from `caption` and `caption_entities` by the fact, that you can set a default value for it via `Defaults.parse_mode` (maybe also have a look at the wiki page on [adding defaults](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Adding-defaults-to-your-bot). Setting default values for `caption` or `caption_entities` doesn't make much sense. So only for `parse_mode` it's relevant to differentviate between 

* `send_media_group(…)` - where `Defaults.parse_mode` must be used
* and `send_media_group(…, parse_mode=None)` - where a `parse_mode` musn't be used

Reply:
Aah, I understand the difference now, thanks! Added commit 4e8007dd766fcdffdf57c6572fcb7f50f9f56633. I hope it's OK to just set `parse_mode` to `DEFAULT_NONE` in the unit test.